### ISSUE TITLE
Clean up gradle build files

### ISSUE THREAD

**Issue by sghpjuikit** [sghpjuikit/player#139]
This is umbrella issue for all gradle-project-structure things.

1 Having gradle build files in one location is against convention. Do we move build files? Where?

2 The first improvement I suggest is to create a player subproject to let the root project encapsulate the entire project and nothing else. Right now `project.gradle.kts` fulfills 2 roles.
 So:
- `project.gradle.kts` becomes place to setup common build configuration, like `allprojects {} `
- `root.gradle.kts` becomes place to set up player module exclusively

**Comment by Xerus2000** [comment]
Please do not call it "root" but "player", root is ambiguous as we.

**Comment by sghpjuikit** [comment]
Blocked by creating buildSrc build, which will split off utility portion of the builds into standalone build, where custom tasks will also reside.

**Comment by sghpjuikit** [comment]
I attempted to split the build file but failed to work it out. I'm putting this off for now. Very low priority.

**Comment by Xerus2000** [comment]
Yeah, things like these are what I do when I get bored, so I'd assign this to me and do it sometime 😉

**Comment by sghpjuikit** [comment]
Thx

**Comment by sghpjuikit** [comment]
I realized you have put certain tasks (buil, run, ...) into a main group. Is there any reason for this other than covnenience?

My real question however is that assuming we split the main build file into that for allprojects and main module, which module should have application plugin and the run task?

**Comment by Xerus2000** [comment]
I would put the run task into the root file, or have it in the player and then add a run task to the root that delegates to the player, kind of a taste thing

**Comment by sghpjuikit** [comment]
I see. So I was on the right track, just lacked the skill to finish through.

**Comment by sghpjuikit** [comment]
kotlinc task that downloads kotlin compiler is now removed. This is handled in runtime by the application itself. See #174

**Comment by sghpjuikit** [comment]
Closing as non-issue. Any improvements in build will be done when needed.

**Comment by sghpjuikit** [comment]
I used `idea` gradle plugin to set up `out` as an `excluded` directory for widget modules. This solves some of the IDE warnings, like class already declared.
